### PR TITLE
Feature/MP-9

### LINKED ISSUE #12: MP-9 api статьи
- Создать три http запроса ([описание запросов](https://github.com/varya-kot/mambet-pages/wiki/API-‐-Статьи))
  - GET /api/v1/articles?page=&size=
  - POST /api/v1/articles
  - PUT /api/v1/articles/{id}
- Запросы должны возвращать mock данные
- Добавить описание запросов в Swagger docs

### PR DESCRIPTION
**Pull request by yanmayak**
Completed first and second endpoint of issue, need reviev of POST /api/v1/articles to complete the task
@ArtemGet @varya-kot

### CODE REVIEW

File: src/main/java/ru/mambetpages/controller/ArticleController.java [hunk 1]
```diff
@@ -41,4 +46,65 @@ public GetArticleDto getArticle(@Parameter(description = "id статьи",
...
+        secondAuthorDto.setName("Игорь");
+        secondAuthorDto.setLastName("Цыба");
+        secondAuthorDto.setId(UUID.randomUUID());
+        secondAuthorDto.setPhoto("Фото");
+
+        firstArticle.setId(UUID.randomUUID());
+        firstArticle.setTitle("Название первой статьи");
+        firstArticle.setContent("Содержание первой статьи");
+        firstArticle.setImage("Изображение первой статьи");
+        firstArticle.setTags(List.of("Тег первой статьи", "Еще один тег первой статьи", "Третий тег первой статьи", "Последний тег первой статьи"));
```
Comment (ArtemGet) [inline src/main/java/ru/mambetpages/controller/ArticleController.java]: use propper formatting:
https://www.yegor256.com/2014/10/23/paired-brackets-notation.html

File: src/main/java/ru/mambetpages/dto/PostArticleDto.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
+package ru.mambetpages.dto;
+
+import io.swagger.v3.oas.annotations.media.Schema;
+import lombok.Data;
+import java.util.List;
+
+@Data
+public class PostArticleDto {
+    @Schema(description = "Название статьи")
+    private String title;
```
Comment (ArtemGet) [inline src/main/java/ru/mambetpages/dto/PostArticleDto.java]: title, content and tags should not be empty or null. 
Just add @Valid to the controller's attributes and other validation annotations to the DTO's attributes.
 
see this article https://www.baeldung.com/spring-boot-bean-validation

**Review by ArtemGet — CHANGES_REQUESTED** [review]
PUT /api/v1/articles/{id} is missing

File: src/main/java/ru/mambetpages/controller/ArticleController.java [hunk 3]
```diff
@@ -41,4 +46,65 @@ public GetArticleDto getArticle(@Parameter(description = "id статьи",
...
+        articleDto.setArticles(articles);
+        articleDto.setPagesCount(76);
+
+        return articleDto;
+    }
+
+    @PostMapping
+    @Operation(summary = "Создание новой статьи")
+    public PostArticleDto postArticle(@RequestBody PostArticleDto articleDto) {
+        PostArticleDto publishArticle = new PostArticleDto();
```
Comment (VaryaGet) [inline src/main/java/ru/mambetpages/controller/ArticleController.java]: This method should return 201 status code with path to suitable GET method , without body 
see http://www.java2s.com/example/java-api/org/springframework/http/responseentity/created-1-2.html

File: src/main/java/ru/mambetpages/controller/ArticleController.java [hunk 4]
```diff
@@ -41,4 +46,65 @@ public GetArticleDto getArticle(@Parameter(description = "id статьи",
...
+
+        articleDto.setArticles(articles);
+        articleDto.setPagesCount(76);
+
+        return articleDto;
+    }
+
+    @PostMapping
+    @Operation(summary = "Создание новой статьи")
+    public PostArticleDto postArticle(@RequestBody PostArticleDto articleDto) {
```
Comment (VaryaGet) [inline src/main/java/ru/mambetpages/controller/ArticleController.java]: For future, return from methods only Get...Dto

**Review by VaryaGet — CHANGES_REQUESTED** [review]

File: src/main/java/ru/mambetpages/dto/PostArticleDto.java [hunk 5]
```diff
@@ -0,0 +1,24 @@
+package ru.mambetpages.dto;
+
+import io.swagger.v3.oas.annotations.media.Schema;
+import jakarta.validation.constraints.NotBlank;
+import lombok.Data;
+import java.util.List;
+
+@Data
+public class PostArticleDto {
```
Comment (VaryaGet) [inline src/main/java/ru/mambetpages/dto/PostArticleDto.java]: rename ArticlePostDto

File: src/main/java/ru/mambetpages/dto/ArticleGetShortDto.java [hunk 6]
```diff
@@ -7,7 +7,7 @@
 import java.util.UUID;
 
 @Data
-public class GetArticleShortDto {
```
Comment (VaryaGet) [inline src/main/java/ru/mambetpages/dto/ArticleGetShortDto.java]: rename ArticleShortGetDto

**Review by VaryaGet — CHANGES_REQUESTED** [review]

**Review by VaryaGet — APPROVED** [review]

**Review by ArtemGet — APPROVED** [review]

**Pull request merged**